hack/vsa_generation_demo/demo-vsa-generation.sh

```diff
@@ -0,0 +1,545 @@
+    kubectl delete secret public-key --ignore-not-found -n openshift-pipelines 2>/dev/null || true
+    
+    # Remove demo resources
+    kubectl delete -f hack/vsa_generation_demo/vsa-demo-resources.yaml --ignore-not-found 2>/dev/null || true
+    
+    # Remove the generate-vsa Tekton task
+    kubectl delete -f config/base/generate-vsa.yaml --ignore-not-found 2>/dev/null || true
+    
+    # Remove RBAC for task runner
+    kubectl delete -f config/base/task-runner-rbac.yaml --ignore-not-found 2>/dev/null || true
+    
+    # Note: We don't remove Tekton Pipelines as other workloads might be using it
+    
+    # Clean up generated keys
+    DEMO_KEYS_DIR="hack/vsa_generation_demo"
+    rm -f "${DEMO_KEYS_DIR}/vsa-demo-keys.key" "${DEMO_KEYS_DIR}/vsa-demo-keys.pub" 2>/dev/null || true
+    
+    # Clean up temporary files
+    rm -f /tmp/vsa-demo-snapshot.yaml 2>/dev/null || true
+    rm -f /tmp/slsa-provenance-cluster.json 2>/dev/null || true
+    rm -f /tmp/demo-release-plan.yaml 2>/dev/null || true
+    
+    # Clean up Docker registry and images
+    if [ "${USE_CLUSTER_REGISTRY:-false}" = "true" ]; then
+        # Clean up port-forward
+        if [ -f /tmp/vsa-demo-port-forward.pid ]; then
+            PORT_FORWARD_PID=$(cat /tmp/vsa-demo-port-forward.pid)
+            kill "$PORT_FORWARD_PID" 2>/dev/null || true
+            rm -f /tmp/vsa-demo-port-forward.pid
+        fi
+        pkill -f "kubectl.*port-forward.*registry.*5001:5000" 2>/dev/null || true
+        
+        # Clean up in-cluster registry
+        kubectl delete -f hack/vsa_generation_demo/in-cluster-registry.yaml --ignore-not-found 2>/dev/null || true
+        docker rmi "${EXTERNAL_REGISTRY:-localhost:5001}/${IMAGE_NAME}:${IMAGE_TAG:-latest}" 2>/dev/null || true
+    else
+        # Clean up localhost registry
+        docker stop vsa-demo-registry 2>/dev/null || true
+        docker rm vsa-demo-registry 2>/dev/null || true
+        docker rmi "${LOCAL_REGISTRY:-localhost:5001}/${IMAGE_NAME}:${IMAGE_TAG:-latest}" 2>/dev/null || true
+    fi
+    
+    # Clean up old TaskRuns created by this demo
+    kubectl get taskruns -n default --no-headers -o custom-columns=":metadata.name,:metadata.creationTimestamp" 2>/dev/null | \
+        awk '$2 < "'$(date -d '1 hour ago' -u +%Y-%m-%dT%H:%M:%SZ)'" {print $1}' | \
+        grep -E '^verify-enterprise-contract-' | \
+        head -5 | \
+        xargs -r kubectl delete taskrun -n default 2>/dev/null || true
+    
+    echo "  Generation demo cleanup completed"
+}
+
+# Set up signal handlers for graceful cleanup
+trap cleanup_generation_demo EXIT
+trap 'echo ""; echo "🛑 Demo interrupted - cleaning up..."; cleanup_generation_demo; exit 1' INT TERM
+
+echo "🚀 VSA Generation Demo - Complete Workflow"
+echo "Working from: $(pwd)"
+echo "=========================================="
+echo ""
+
+# Clean up any conflicting resources from previous demos
+echo "🔄 Cleaning up any existing demo resources..."
```

Comment:
Should this cleanup be in the `cleanup_generation_demo` function?

Reply:
Agreed. Addressed in a follow-up.